File: app-service/src/main/java/com/appverse/app_service/dto/ApplicationResponse.java
// === In app-service Project ===
package com.appverse.app_service.dto;

import com.appverse.app_service.enums.MonetizationType; // <<< IMPORT ADDED
import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

public record ApplicationResponse(
    String id,
    String name,
    String tagline,
    String description,
    String version,
    String categoryId,
    BigDecimal price,
    String currency,
    boolean isFree,

    // ****************************************************
    // ****** CHANGES START HERE FOR RESPONSE DTO *******
    // ****************************************************
    MonetizationType monetizationType,
    List<String> associatedSubscriptionPlanIds,
    // ****************************************************
    // ****** CHANGES END HERE FOR RESPONSE DTO *********
    // ****************************************************

    List<String> platforms,
    String accessUrl,
    String websiteUrl,
    String supportUrl,
    String thumbnailUrl,
    List<ScreenshotResponse> screenshots,
    String developerId,
    String developerName,
    String categoryName,
    List<String> tags,
    String status,
    Instant publishedAt,
    Instant createdAt,
    Instant updatedAt,
    Double averageRating,
    Integer ratingCount
) {}

File: frontend/appverse/src/app/features/application/models/application-response.ts
// src/app/models/application-response.ts

import { ApplicationStatus } from "./application-status";

export interface ApplicationResponse {
  id: string;
  name: string;
  tagline: string;
  description: string;
  version: string;
  categoryId: string;
  price: number; // BigDecimal -> number in TS
  currency: string;
  isFree: boolean;
  monetizationType: 'FREE' | 'PAID' | 'SUBSCRIPTION'; // Adjust based on your enum values
  associatedSubscriptionPlanIds: string[];
  platforms: string[];
  accessUrl: string;
  websiteUrl: string;
  supportUrl: string;
  thumbnailUrl: string;
  screenshots: ScreenshotResponse[];
  developerId: string;
  developerName: string;
  categoryName: string;
  tags: string[];
  status: ApplicationStatus;
  publishedAt: string; // Instant -> ISO string
  createdAt: string; // Instant -> ISO string
  updatedAt: string; // Instant -> ISO string
  averageRating: number;
  ratingCount: number;
}

export interface ScreenshotResponse {
  id: string;
  filename: string;
  url: string;
  order: number;
  caption?: string;
}
